sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -83,12 +99,147 @@ internal class AmqpMessageConverter
+        ///   The caller is assumed to hold ownership over the specified message, including
+        ///   ensuring proper disposal.
+        /// </remarks>
+        ///
         public virtual EventData CreateEventFromMessage(AmqpMessage source)
         {
             Guard.ArgumentNotNull(nameof(source), source);
             return BuildEventFromAmqpMessage(source);
         }
 
+        /// <summary>
+        ///   Creates an <see cref="AmqpMessage" /> for use in requesting the properties of an Event Hub.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub to query the properties of.</param>
+        /// <param name="managementAuthorizationToken">The bearer token to use for authorization of management operations.</param>
+        ///
+        /// <returns>The AMQP message for issuing the request.</returns>
+        ///
+        /// <remarks>
+        ///   The caller is assumed to hold ownership over the message once it has been created, including
+        ///   ensuring proper disposal.
+        /// </remarks>
+        ///
+        public virtual AmqpMessage CreateEventHubPropertiesRequest(string eventHubName,
+                                                                   string managementAuthorizationToken)
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(eventHubName), eventHubName);
+            Guard.ArgumentNotNullOrEmpty(nameof(managementAuthorizationToken), managementAuthorizationToken);
+
+            var request = AmqpMessage.Create();
+            request.ApplicationProperties = new ApplicationProperties();
+            request.ApplicationProperties.Map[AmqpManagement.ResourceNameKey] = eventHubName;
+            request.ApplicationProperties.Map[AmqpManagement.OperationKey] = AmqpManagement.ReadOperationValue;
+            request.ApplicationProperties.Map[AmqpManagement.ResourceTypeKey] = AmqpManagement.EventHubResourceTypeValue;
+            request.ApplicationProperties.Map[AmqpManagement.SecurityTokenKey] = managementAuthorizationToken;
+
+            return request;
+        }
+
+        /// <summary>
+        ///   Converts a given <see cref="AmqpMessage" /> response to a query for Event Hub properties into the
+        ///   corresponding <see cref="EventHubProperties" /> representation;
+        /// </summary>
+        ///
+        /// <param name="response">The response message to convert.</param>
+        ///
+        /// <returns>The set of properties represented by the response.</returns>
+        ///
+        /// <remarks>
+        ///   The caller is assumed to hold ownership over the specified message, including
+        ///   ensuring proper disposal.
+        /// </remarks>
+        ///
+        public virtual EventHubProperties CreateEventHubPropertiesFromResponse(AmqpMessage response)
+        {
+            Guard.ArgumentNotNull(nameof(response), response);
+
+            var responseData = response?.ValueBody?.Value as AmqpMap;
```

Comment:
Is the null conditional required here on `response` if there is a guard check previously?

Reply:
The response is returned by the AMQP link.   According to the equivalent code in Track One, it can be null in some failure conditions.  Since this is not a user-provided value, we don't want to present with an argument exception.  We want to continue down the error path and return something meaningful in the AMQP context.

Reply:
Per offline conversation, we're not in the area that I thought we were.  You're correct that it's not needed here.  Removed.